eng/service.proj

```diff
@@ -71,4 +72,24 @@
             SkipNonexistentProjects="false"
             SkipNonexistentTargets="true" />
   </Target>
+  <Target Name="ProjectDependsOn">
+    <MSBuild Projects="@(ProjectReference)"
+            Targets="ProjectDependsOn"
+            BuildInParallel="$(BuildInParallel)"
+            SkipNonexistentProjects="false"
+            SkipNonexistentTargets="true">
+      <Output ItemName="ProjectsToInclude" TaskParameter="TargetOutputs"/>
+    </MSBuild>
+
+    <RemoveDuplicates Inputs="@(ProjectsToInclude)">
+      <Output TaskParameter="Filtered" ItemName="ProjectsToIncludeFitered"/>
+    </RemoveDuplicates>
+    <Message Text="%(ProjectsToIncludeFitered.Identity)" Importance="High" />
+    <ItemGroup>
+       <_LinesToWrite Include="%3CProject%3E%0A%20%20%3CItemGroup%3E" />
+       <_LinesToWrite Include="%20%20%20%20%3CProjectReference%20Include%3D'%(ProjectsToIncludeFitered.Identity)'%20%2F%3E" />
```

Comment:
We can simply output the project list directly and avoid the xml if you are going to parse this outside of msbuild.